ui/src/components/comment-node.tsx

```diff
@@ -160,6 +162,12 @@ export class CommentNode extends Component<CommentNodeProps, CommentNodeState> {
                   }}
                 />
               </span>
+
+              <CakeDay
+                creator_name={creator_name}
+                creator_published={creator_published}
+              />
```

Comment:
I know it sounds strange and less than ideal, but I'd rather you move the cakeday check to `utils.ts`, or import the check from your cakeday component and here do a 
`isCakeDay(creator_published) && <CakeDay...`

The reason is, I don't want to hide conditionals inside of components, I want it to be apparent here, that `<CakeDay` will only be rendered if it passes the check.

I know its not as clean, but otherwise it makes it seem that its showing cakeday no matter what, only to discover that its not, after you go inside the component.

Reply:
> The reason is, I don't want to hide conditionals inside of components, I want it to be apparent here, that <CakeDay will only be rendered if it passes the check.

> I know its not as clean, but otherwise it makes it seem that its showing cakeday no matter what, only to discover that its not, after you go inside the component.

Alright, this sort of makes sense I can see your point.

I have kept the `CakeDay` component but moved `isCakeDay` to the `utils.ts`. 
I think that this function may perhaps may be reused in future in some other components so it may be useful to have it there.

Changed in 8fe8836bc263be11895fafa958409499c9e9fc9d.